### PR TITLE
Refine logging in pkg/bpm, move package level variable log into struct BackgroundProcessManager and ProcessBuilder as a field

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
#2870
### What's changed and how it works?

Use NewDefaultZapLogger.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/bpm/bpm.go [hunk 1]
```diff
@@ -76,20 +77,25 @@ type Stdio struct {
...
 type BackgroundProcessManager struct {
-	deathSig    *sync.Map
-	identifiers *sync.Map
-	stdio       *sync.Map
-
+	deathSig         *sync.Map
+	identifiers      *sync.Map
+	stdio            *sync.Map
 	metricsCollector *metricsCollector
+	log              logr.Logger
```
Comment (STRRL) [inline pkg/bpm/bpm.go:84]: please use `logger` here.

File: pkg/bpm/bpm.go [hunk 2]
```diff
@@ -76,20 +77,25 @@ type Stdio struct {
...
 }
 
 // NewBackgroundProcessManager creates a background process manager
 func NewBackgroundProcessManager(registry prometheus.Registerer) BackgroundProcessManager {
+	rootLogger, err := clog.NewDefaultZapLogger()
+	if err != nil {
+		stdlog.Fatal("failed to create root logger", err)
+	}
```
Comment (STRRL) [inline pkg/bpm/bpm.go:92]: please do not create another logger, acquire the logger as a parameter in the constructor.

File: pkg/bpm/bpm.go [hunk 3]
```diff
@@ -375,13 +381,18 @@ func (m *BackgroundProcessManager) GetIdentifiers() []string {
 
 // DefaultProcessBuilder returns the default process builder
 func DefaultProcessBuilder(cmd string, args ...string) *ProcessBuilder {
+	rootLogger, err := clog.NewDefaultZapLogger()
+	if err != nil {
+		stdlog.Fatal("failed to create root logger", err)
+	}
```
Comment (STRRL) [inline pkg/bpm/bpm.go:387]: ditto, please do not create another logger, acquire the logger as a parameter in the constructor.

**Review by STRRL — CHANGES_REQUESTED** [review]

**Comment by mikechengwei** [comment]
@STRRL  I have reoptimize the code , plz help me review this pr again. Thks.

File: cmd/chaos-daemon/main.go [hunk 4]
```diff
@@ -58,7 +58,13 @@ func main() {
 		os.Exit(0)
 	}
 
-	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	ctrl.Log.WithName("chaos-daemon")
```
Comment (STRRL) [inline cmd/chaos-daemon/main.go:61]: line 61 could be deleted

File: cmd/chaos-daemon/main.go [hunk 5]
```diff
@@ -67,15 +73,15 @@ func main() {
...
-	err := fusedev.GrantAccess()
+	rootLogger.Info("grant access to /dev/fuse")
+	err = fusedev.GrantAccess()
 	if err != nil {
-		log.Error(err, "grant access to /dev/fuse")
+		rootLogger.Error(err, "grant access to /dev/fuse")
 	}
 
-	server, err := chaosdaemon.BuildServer(conf, reg)
+	server, err := chaosdaemon.BuildServer(conf, reg, rootLogger)
```
Comment (STRRL) [inline cmd/chaos-daemon/main.go:82]:
```suggestion
	server, err := chaosdaemon.BuildServer(conf, reg, rootLogger.WithName("chaos-daemon"))
```

File: pkg/chaosdaemon/server.go [hunk 6]
```diff
@@ -82,25 +83,27 @@ type DaemonServer struct {
 
 	IPSetLocker      *locker.Locker
 	TimeChaosManager tasks.TaskManager
+	logger           logr.Logger
```
Comment (STRRL) [inline pkg/chaosdaemon/server.go:86]: This logger seems not used at anywhere, we could remove it.

**Review by STRRL — COMMENTED** [review]
rest LGTM!

**Comment by mikechengwei** [comment]
This pr have finished.

**Pull request closed without merging**